Code/GraphMol/RGroupDecomposition/RGroupScore.cpp

```diff
@@ -38,20 +38,34 @@ double matchScore(const std::vector<size_t> &permutation,
       << std::endl;
   }
 #endif
-  // For each label (group)
+
+  // What is the largest rgroup count at any label
+  int N = 0;
+  std::map<int, int> num_rgroups;
+  for (size_t m = 0; m < permutation.size(); ++m) {  // for each molecule
+  for (auto l : matches[m][permutation[m]].rgroups) {
+    int count = ++num_rgroups[l.first];
+    if (N < count) {
+        N = count;
+      }
+    }
+  }
+  // for each label (r-group)
   for(auto l : labels ) {
 #ifdef DEBUG
     std::cerr << "Label: " << l << std::endl;
 #endif
     std::vector<std::map<std::string, unsigned int>> matchSetVect;
     std::map<std::set<int>, size_t> linkerMatchSet;
 
+    int num_rgroups_for_label = 0;
     for (size_t m = 0; m < permutation.size(); ++m) {  // for each molecule
 
       auto rg = matches[m][permutation[m]].rgroups.find(l);
       if (rg == matches[m][permutation[m]].rgroups.end()) {
         continue;
       }
+      num_rgroups_for_label++;
```

Comment:
It looks like this is only being used inside the `#ifdef DEBUG`. Maybe worth either removing or doing:
```suggestion
#ifdef DEBUG
      num_rgroups_for_label++;
#endif
```

Reply:
It is being used in the matchSet loop.

Reply:
sorry, missed that.